src/track/beats.cpp

```diff
@@ -802,4 +780,39 @@ void Beats::slotTrackBeatsUpdated() {
     m_beatsInternal.setDurationSeconds(m_track->getDuration());
 }
 
+int BeatsInternal::numBeatsInRange(FramePos startFrame, FramePos endFrame) const {
+    FramePos lastCountedBeat(0.0);
+    int iBeatsCounter;
+    for (iBeatsCounter = 1; lastCountedBeat < endFrame; iBeatsCounter++) {
+        lastCountedBeat = findNthBeat(startFrame, iBeatsCounter);
+        if (lastCountedBeat == kInvalidFramePos) {
+            break;
+        }
+    }
+    return iBeatsCounter - 2;
```

Comment:
Can this return a negative value?

Reply:
I think it could if `lastCountedBeat == kInvalidFramePos` on the first iteration. The loop would break before `iBeatsCounter++`.